### ISSUE TITLE
Parametrizing node and channel parameters

### ISSUE THREAD

**Issue by chaseking** [jaxleyverse/jaxley#602]
In fitting biophysical models with many nodes and channels the parameter count can quickly become large. It would be useful to have some constraints on this distribution of parameters—for example, a channel conductance value being a scaled/shifted sigmoid function of the distance from the node to the soma, and these sigmoid parameters are learned through optimization. This could reduce the number of parameters by many orders of magnitude and (perhaps) make fitting easier.

I can't seem to find a way to do this with the current API. Is this possible right now? If so, how can I go about achieving this? Thanks!

**Comment by michaeldeistler** [comment]
Hi there!

Thanks for reaching out! There is no clean API for this at this point. I think it's worth thinking about making this more easily possible, but for the time being you will need a few more lines of code for this :)

I think the following should work:
```python
import jax.numpy as jnp
import jaxley as jx
from jaxley.channels import Leak


comp = jx.Compartment()
branch = jx.Branch(comp, ncomp=2)
cell = jx.Cell(branch, parents=[-1, 0, 0])
cell.insert(Leak())
cell.compute_xyz()

cell[0, 0].stimulate(0.1 * jnp.ones((100,)))
cell[0, 0].record()

def simulate(sigmoid_offset, global_scaling):
    pstate = None
    for branch in cell:
        for comp in branch:
            distance = cell[0, 0].distance(comp)
            conductance = global_scaling / (jnp.exp(-(distance + sigmoid_offset)))
            comp.data_set("Leak_gLeak", conductance, pstate)
    return jx.integrate(cell, param_state=pstate)

sigmoid_offset = jnp.ones((1,))
global_scaling = jnp.ones((1,))
v = simulate(sigmoid_offset, global_scaling)
```

Does this help?
Michael

**Comment by michaeldeistler** [comment]
Alright, I just saw that this cannot be jitted...I think [this PR](https://github.com/jaxleyverse/jaxley/pull/603) will fix it, but I have to run a few more tests to make sure.

**Comment by chaseking** [comment]
Cheers Michael, appreciate the quick reply! This looks good and I'll let you know if anything else comes up.

**Comment by michaeldeistler** [comment]
Closing this as #603 is merged. Will release later today. Thanks for raising this!
